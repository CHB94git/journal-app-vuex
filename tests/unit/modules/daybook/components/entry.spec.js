import { shallowMount } from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry';
import { journalState } from '../../../mocks/test-journal-state';

describe('Pruebas componente entry', () => {
    let wrapper

    const mockRouter = {
        push: jest.fn()
    }

    beforeEach(() => {
        wrapper = shallowMount(Entry, {
            props: {
                entry: journalState.entries[0]
            },
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })
    });


    it('debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });


    it('debe de redireccionar al hacer clic en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')

        entryContainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()

        expect(mockRouter.push).toHaveBeenCalledWith(
            {
                name: 'entry',
                params: {
                    // '-N6Qv2Q_OCTqBCWgCnME'
                    id: journalState.entries[0].id
                }
            }
        )
    });


    it('pruebas en las propiedades computadas', () => {
        expect(wrapper.vm.day).toBe(7);
        expect(wrapper.vm.month).toBe('Julio');
        expect(wrapper.vm.yearDay).toBe('2022, Jueves');
    });
});