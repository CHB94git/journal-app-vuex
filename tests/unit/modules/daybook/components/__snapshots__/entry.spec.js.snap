// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente entry debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Título -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">7</span><span class="mx-1 fs-5">Julio</span><span class="mx-2 fw-light">2022, Jueves</span></div><!-- Descripción -->
  <div class="entry-description">Hoy estoy aprendiendo Vue.js con Fernando Herrera</div>
</div>
`;
